core/src/main/java/org/elasticsearch/cluster/AbstractNamedDiffable.java

```diff
@@ -104,6 +113,12 @@ public T apply(T part) {
         public String getWriteableName() {
             return name;
         }
+
+        @Override
+        public Version getMinimalSupportedVersion() {
+            assert minimalSupportedVersion != null;
```

Comment:
I think it is a bit weird to declare this as nullable but assert that it isn't null in a getter. Maybe it'd be better if you put a message here and on the other assertions like "getMinimalSupportedVersions should only be called on the write side but looks like it was called on the read side because minimalSupportedVersion is null".